handler/pullRequestHandler.go

```diff
@@ -101,6 +86,57 @@ That's something we need before your Pull Request can be merged. Please see our
 	}
 }
 
+// VerifyPullRequestDescription checks that the PR has anything in the body.
+// If there is no body, an 'invalid' label is added and comment posted to the PR with a link to the contributing guide.
+func VerifyPullRequestDescription(req types.PullRequestOuter, contributingURL string, config config.Config) {
+	ctx := context.Background()
+	token := getAccessToken(config, req.Installation.ID)
+	client := factory.MakeClient(ctx, token, config)
+
+	if req.Action == "opened" {
+		if missingDescription(req.PullRequest) {
+			fmt.Println("Applying invalid label")
+			_, res, assignLabelErr := client.Issues.AddLabelsToIssue(ctx, req.Repository.Owner.Login, req.Repository.Name, req.PullRequest.Number, []string{"invalid"})
+			if assignLabelErr != nil {
+				log.Fatalf("%s limit: %d, remaining: %d", assignLabelErr, res.Limit, res.Remaining)
+			}
+
+			body := `Thank you for your contribution. I've just checked and your Pull Request doesn't appear to have any description.
+That's something we need before your Pull Request can be merged. Please see our [contributing guide](` + contributingURL + `).`
+
+			comment := &github.IssueComment{
+				Body: &body,
+			}
+
+			comment, resp, err := client.Issues.CreateComment(ctx, req.Repository.Owner.Login, req.Repository.Name, req.PullRequest.Number, comment)
+			if err != nil {
+				log.Fatalf("%s limit: %d, remaining: %d", assignLabelErr, resp.Limit, resp.Remaining)
+				log.Fatal(err)
+			}
+			fmt.Println(comment, resp.Rate)
+		}
+	}
+}
+
+func getAccessToken(config config.Config, installationID int) string {
+	token := os.Getenv("personal_access_token")
+	if len(token) == 0 {
+
+		newToken, tokenErr := auth.MakeAccessTokenForInstallation(
+			config.ApplicationID,
+			installationID,
+			config.PrivateKey)
+
+		if tokenErr != nil {
```

Comment:
probably better to bubble up an error and handle properly than blow up. I know this may have been refactored out, but that now means it's getting eyes on it. My hope is to remove all `log.Fatal*` calls and `panic` so we can use Derek as an SDK in the future.

Reply:
I wasn't quite sure how the error should be handled. For now, just writing the error to stdout and returning before the token is used to prevent a panic

Reply:
How about returning an error from the function?